enhancements/authentication/pod-security-admission-enforcement.md

```diff
@@ -0,0 +1,428 @@
+
+- With the `security.openshift.io/MinimallySufficientPodSecurityStandard` annotation, it can evaluate Namespaces that lack the `pod-security.kubernetes.io/enforce` label but have user-overridden warn or audit labels.
+- With `ValidatedSCCSubjectType`, the controller can classify issues arising from user-based SCC workloads separately.
+  Many of the remaining clusters with violations appear to involve workloads admitted by user SCCs.
+
+### Secure Rollout
+
+The Proposal section indicates that enforcement will be introduced first at the Namespace level and later at the global (cluster-wide) level.
+In addition to adjusting how the `OpenShiftPodSecurityAdmission` `FeatureGate` behaves, administrators need visibility and control throughout this transition.
+A new API is necessary to provide this flexibility.
+
+#### New API
+
+This API offers a gradual way to roll out Pod Security Admission enforcement to clusters.
+It gives users the ability to influence the rollout and see feedback on which Namespaces might violate Pod Security standards.
+
+```go
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PSAEnforcementMode indicates the actual enforcement state of Pod Security Admission
+// in the cluster. Unlike PSATargetMode, which reflects the user’s desired or “target”
+// setting, PSAEnforcementMode describes the effective mode currently active.
+//
+// The modes define a progression from no enforcement, through label-based enforcement
+// to label-based with global config enforcement.
+type PSAEnforcementMode string
+
+const (
+	// EnforcementModePrivileged indicates that no Pod Security restrictions
+	// are effectively applied.
+	// This aligns with a pre-rollout or fully "privileged" cluster state,
+	// where neither enforce labels are set nor the global config enforces "Restricted".
+	EnforcementModePrivileged PSAEnforcementMode = "Privileged"
+
+	// EnfrocementModeLabel indicates that the cluster is enforcing Pod Security
+	// labels at the Namespace level (via the PodSecurityAdmissionLabelSynchronizationController),
+	// but the global kube-apiserver configuration is still "Privileged."
+	EnfrocementModeLabel PSAEnforcementMode = "LabelEnforcement"
+
+	// EnforcmentModeFull indicates that the cluster is enforcing
+	// labels at the Namespace level, and the global configuration has been set
+	// to "Restricted" on the kube-apiserver.
+	// This represents full enforcement, where both Namespace labels and the global config
+	// enforce Pod Security Admission restrictions.
+	EnforcmentModeFull PSAEnforcementMode = "FullEnforcement"
+)
+
+// PSATargetMode reflects the user’s chosen (“target”) enforcement level.
+type PSATargetMode string
+
+const (
+	// TargetModePrivileged indicates that the user wants no Pod Security
+	// restrictions applied. The desired outcome is that the cluster remains
+	// in a fully privileged (pre-rollout) state, ignoring any label enforcement
+	// or global config changes.
```

Comment:
Would this mode actually ignore label enforcement? 

If I am following https://kubernetes.io/docs/tasks/configure-pod-container/enforce-standards-admission-controller/#configure-the-admission-controller correctly, it seems like the defaults only apply when there are no labels. Is `Privileged` effectively the same as disabling the `PodSecurity` admission plugin?

Reply:
If `TargetModePrivileged` is set, based on what I tried to articulate, is that we effectively disable  any restrictions, which means it should behave as it does in all cluster before.

In detail, we could argue about how we would like to do it. I would just revert everything, like removing the enforce label set by the PSA label syncer and set the global config back to privileged.
Logically (based on the name) more correct would be to set the enforce label to privileged. But as mentioned above, I would like to keep the permutations of possible state as minimal as possible.

Reply:
For the "I don't want Pod Security at all", maybe something like `None` is more reflective of that and in that case should disable the Pod Security admission plugin all together?